CredentialProvider.Microsoft/RequestHandlers/GetAuthenticationCredentialsRequestHandler.cs

```diff
@@ -78,6 +67,16 @@ public override async Task<GetAuthenticationCredentialsResponse> HandleRequestAs
                     continue;
                 }
 
+                if (credentialProvider.IsCachable && TryCache(request, out string cachedToken))
```

Comment:
We can store the output for TryCache() outside the for loop and use the result inside. Saves us some calls to cache.

Reply:
We only want to try cache if the cred provider is cachable and we need to check it inside the loop. Currently only 1 out of 3 is cachable. With build task cred providers we never want to try cache.

Reply:
ah ok. makes sense then.